sklearn/calibration.py

```diff
@@ -33,25 +33,31 @@
 from .utils.validation import _deprecate_positional_args
 
 
-def _fit_calibrated_classifer(estimator, X, y, train, test, supports_sw,
-                              method, classes, sample_weight=None):
+def _fit_calibrated_classifer(estimator, X, y, train, test,
+                              method, classes, sample_weight=None,
+                              base_estimator_uses_sw=False, **fit_params):
     """Fit calibrated classifier for a given dataset split.
 
     Returns
     -------
     calibrated_classifier : estimator object
         The calibrated estimator.
     """
-    if sample_weight is not None and supports_sw:
-        estimator.fit(X[train], y[train],
-                      sample_weight=sample_weight[train])
+    fit_params_train = {}
+    for key, val in fit_params.items():
+        fit_params_train[key] = _safe_indexing(val, train)
+
+    if base_estimator_uses_sw:
+        swbe = _safe_indexing(sample_weight, train)
+        estimator.fit(
+            X[train], y[train], sample_weight=swbe, **fit_params_train)
     else:
-        estimator.fit(X[train], y[train])
+        estimator.fit(X[train], y[train], **fit_params_train)
 
     calibrated_classifier = _CalibratedClassifier(estimator,
                                                   method=method,
                                                   classes=classes)
-    sw = None if sample_weight is None else sample_weight[test]
+    sw = None if sample_weight is None else _safe_indexing(sample_weight, test)
     calibrated_classifier.fit(X[test], y[test], sample_weight=sw)
```

Comment:
we could actually split the data above in the same if/else

Reply:
I put it in the snippet

Reply:
:+1: